cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -317,7 +318,8 @@ private void recordCompletedWorkflow(final Id.Workflow id, String pid) {
     Map<String, WorkflowNode> nodeIdMap = workflowSpec.getNodeIdMap();
     final List<WorkflowDataset.ProgramRun> programRunsList = new ArrayList<>();
     for (Map.Entry<String, String> entry : run.getProperties().entrySet()) {
-      if (!("workflowToken".equals(entry.getKey()) || "runtimeArgs".equals(entry.getKey()))) {
+      if (!("workflowToken".equals(entry.getKey()) || "runtimeArgs".equals(entry.getKey())
+        || "workflowNodeState".equals(entry.getKey()))) {
```

Comment:
This logic needs to be refactored. Filed a JIRA - https://issues.cask.co/browse/CDAP-5319